src/frontend/utils/resumeLive.ts

```diff
@@ -0,0 +1,41 @@
+import { Parser } from 'm3u8-parser';
+
+import { getResource } from '../data/sideEffects/getResource';
+import { modelName } from 'types/models';
+import { Video } from '../types/tracks';
+
+const regex = /^(https.*)\/.*\.m3u8$/;
+
+export const resumeLive = async (video: Video) => {
+  const manifestUrl = video.urls?.manifests.hls!;
+
+  const mainManifest = await fetchMainManifest(manifestUrl);
+  const firstManifest = mainManifest.playlists[0].uri;
+  const matches = manifestUrl.match(regex);
+
+  await pollEndedManifest(`${matches![1]}/${firstManifest}`);
+  await getResource(modelName.VIDEOS, video.id);
+};
+
+const pollEndedManifest = async (manifestUrl: string) => {
+  const manifest = await (await fetch(manifestUrl)).text();
+  const hlsParser = new Parser();
+  hlsParser.push(manifest);
+  hlsParser.end();
+
+  if (hlsParser.manifest.endList) {
+    await new Promise((resolve) => window.setTimeout(resolve, 2500));
+    await pollEndedManifest(manifestUrl);
+  }
+
+  return null;
+};
+
+const fetchMainManifest = async (manifestUrl: string) => {
+  const manifest = await (await fetch(manifestUrl)).text();
```

Comment:
Not a fan of `await ( await ...`
```suggestion
  const manifest = await fetch(manifestUrl).then(response => response.text());
```

Furthermore, I wonder what would happen if request failed.

Reply:
I will manage the failure where this function is called.